### PR TITLE
BFD-897: Redirect all test output to separate files

### PR DESCRIPTION
**Pull request by karlmdavis**
### Change Details

This was already _mostly_ happening for ITs, but I moved that into the parent POM to apply it everywhere and also turned it on for unit tests. (Surefire runs our unit tests and failsafe runs our ITs.)

Should help make our build logs a bit more manageable, particularly on GH Actions and Jenkins.

In addition:
* Cleaned up the failsafe POM entries a bit, to improve consistency.
* Set GH Actions to save our test data files, e.g. `failsafe-reports`, as build artifacts.

### Acceptance Validation

1. Do you agree this is a good idea?
2. Does it work as expected?

### Feedback Requested

(see above)

### External References

- [BFD-897](https://jira.cms.gov/browse/BFD-897)

### Security Implications

This PR has no security implications.

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Review by cbrunefearless — APPROVED** [review]
LGTM

**Comment by karlmdavis** [comment]
This PR is blocked by #665: its GH Actions build won't succeed until that's been merged in.

**Review by mackec1 — APPROVED** [review]

**Review by brianburton — APPROVED** [review]
Nice.  Adds functionality while simplifying the poms.

**Comment by karlmdavis** [comment]
Note to self: I think I need to split up our GH Actions file to not use a matrix, as the artifacts aren't job-specific, and we don't really want to save any artifacts from the formatter job.

**Review by mackec1 — APPROVED** [review]

**Comment by karlmdavis** [comment]
@jzulim and all: FYI, we'll have to update our branch protections before merging this PR, as it changes the job names.

File: .github/workflows/ci-java.yml
Comment (rtravitz) [inline .github/workflows/ci-java.yml:5]: Why do we use a MacOS runner for the CI job? I see it was that way previously, but would an Ubuntu virtual environment make more sense to keep it closer to how things will run in Jenkins and deployed envs?

Comment (karlmdavis) [reply .github/workflows/ci-java.yml:5]: I believe that we switched to that a while back, to either workaround or reliably reproduce (I don't recall) a platform-specific bug.

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Review by ericbuckley — APPROVED** [review]
👍

**Pull request merged**
